### PR TITLE
[Custom Fields] List empty state

### LINKED ISSUE #14254: Empty state UI

### PR DESCRIPTION
**Pull request by hafizrahman**
Closes: #14254 

## Description

This PR adds the empty state for Custom Fields List screen. The text and design matches the one in Android, except for the image. I can't find the same image in iOS, so I opted for something that looks somewhat close. Please check the comparison in the screenshot area below.

## Steps to reproduce

1. Go to product details of a product with no custom fields (Product Details > Add More Details > Custom Fields)
2. Ensure empty state view is shown like in screenshot below,
3. Test the "Learn more" button, ensure it opens webview correctly to the Custom Fields documentation page in WooCommerce.com,
4. Go back and test adding a new custom field, ensure the empty view is then replaced by the list.

## Testing information

I tested the steps above with iPhone simulator with iOS 17.5.

## Screenshots

| Android light | Android dark |
|-|-|
|  |  |

| iOS light | iOS dark |
|-|-|
| Simulator Screenshot - iPhone 15 Pro - 2024-10-30 at 18 16 12 |  Simulator Screenshot - iPhone 15 Pro - 2024-10-30 at 18 25 26 |

---

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.42

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/230">21.0</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/EmptyState.swift [hunk 1]
```diff
@@ -35,6 +46,7 @@ private extension EmptyState {
         static let verticalSpacing: CGFloat = 16
         static let horizontalSpacing: CGFloat = 24
         static let width: CGFloat = 168
+        static let buttonWidth: CGFloat = 228
```
Comment (hafizrahman) [inline WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/EmptyState.swift:49]: This width is matching the button width on `EmptyStateViewController`:

https://github.com/woocommerce/woocommerce-ios/blob/398e90e0ffe7f9b86bf7dcf5173c1c41220d4e0b/WooCommerce/Classes/ViewRelated/ReusableViews/EmptyStateViewController/EmptyStateViewController.xib#L57

File: WooCommerce/Classes/System/WooConstants.swift [hunk 2]
```diff
@@ -366,6 +366,8 @@ extension WooConstants {
 
         case ordersScreen = "https://woocommerce.com/mobile/orders"
 
+        case customFieldsLearnMore = "https://woocommerce.com/document/custom-product-fields/"
```
Comment (hichamboushaba) [inline WooCommerce/Classes/System/WooConstants.swift]: @hafizrahman I didn't review the PR yet, but I just noticed that we always use the product documentation here, as agreed before (p1727686013865629/1727685068.374159-slack-C03L1NF1EA3), the URL should depend on the `MetaDataType`, and we should use the page https://woocommerce.com/document/managing-orders/view-edit-or-add-an-order/#custom-fields for orders.

Comment (hichamboushaba) [reply WooCommerce/Classes/System/WooConstants.swift]: We already pass the `MetaDataType` to the `CustomFieldsListViewModel`, so it should be easy to use it to have a dynamic URL.

Comment (hafizrahman) [reply WooCommerce/Classes/System/WooConstants.swift]: Good catch. Added in https://github.com/woocommerce/woocommerce-ios/pull/14260/commits/5e29255a927932237a2e71dba8d366ceb63f5f42

Please note that unlike in Android, an existing behavior for the View Custom Fields button in Order Details is that it's hidden by default if the Order has no custom fields:

https://github.com/woocommerce/woocommerce-ios/blob/5a0ddfec620eb33a9c2dafc363e05c88149b8b45/WooCommerce/Classes/ViewModels/Order%20Details/OrderDetailsDataSource.swift#L1235-L1238

So testing the empty state in Order will have to be done by deleting existing custom fields (but without saving the deletion).

Comment (hichamboushaba) [reply WooCommerce/Classes/System/WooConstants.swift]: > Please note that unlike in Android, an existing behavior for the View Custom Fields button in Order Details is that it's hidden by default if the Order has no custom fields:

This behavior was in Android too but I removed it, I think we should do this for iOS too, do you prefer to handle it on this PR or on a separate one?

**Review by hichamboushaba — APPROVED** [review]
Nice work @hafizrahman, it works well.

I'm not merging yet until you check this [point](https://github.com/woocommerce/woocommerce-ios/pull/14260#discussion_r1824075559).

File: WooCommerce/Classes/System/WooConstants.swift (hunk 2 above)
Comment (hafizrahman) [reply WooCommerce/Classes/System/WooConstants.swift]: I'll add it here.

Comment (hafizrahman) [reply WooCommerce/Classes/System/WooConstants.swift]: https://github.com/woocommerce/woocommerce-ios/pull/14260/commits/35f3969ff2a3e6c344d85499d561099446503ea3

**Pull request merged**